source/common/network/address_impl.cc

```diff
@@ -64,6 +64,70 @@ int Ipv4Instance::socket(SocketType type) const {
   return ::socket(AF_INET, flagsFromSocketType(type), 0);
 }
 
+std::string Ipv6Instance::Ipv6Helper::address() const {
+  return std::string(reinterpret_cast<const char*>(address_.sin6_addr.s6_addr),
+                     sizeof address_.sin6_addr.s6_addr);
+}
+
+uint32_t Ipv6Instance::Ipv6Helper::port() const {
+  return ntohs(address_.sin6_port);
+}
+
+std::string Ipv6Instance::Ipv6Helper::makeFriendlyAddress() const {
+  char str[INET6_ADDRSTRLEN];
+  if (nullptr == inet_ntop(AF_INET6, &address_.sin6_addr, str, INET6_ADDRSTRLEN)) {
```

Comment:
I don't think this can actually happen? I would just ASSERT that it returns non-null if you want (might want to fix IPv4 code above to do the same thing).